pkg/resources/configmaps.go

```diff
@@ -0,0 +1,325 @@
+          private_key_path /fluentd/etc/https/tls.key
+        </transport>
+        <parse>
+          @type json
+        </parse>
+    </source>
+    <filter icp-audit>
+        @type parser
+        format json
+        key_name message
+        reserve_data true
+    </filter>
+    <filter icp-audit.*>
+        @type record_transformer
+        enable_ruby true
+        <record>
+          tag ${tag}
+          message ${record.to_json}
+        </record>
+    </filter>`
+
+var splunkConfigData1 = `
+splunkHEC.conf: |-
+    <match icp-audit icp-audit.**>`
+var splunkDefaults = `
+        @type splunk_hec
+        hec_host SPLUNK_SERVER_HOSTNAME
+        hec_port SPLUNK_PORT
+        hec_token SPLUNK_HEC_TOKEN
+        ca_file /fluentd/etc/tls/splunkCA.pem
+        source ${tag}`
+var splunkConfigData2 = `
+    </match>`
+
+var qRadarConfigData1 = `
+remoteSyslog.conf: |-
+    <match icp-audit icp-audit.**>`
+var qRadarDefaults = `
+        @type copy
+        <store>
+            @type remote_syslog
+            host QRADAR_SERVER_HOSTNAME
+            port QRADAR_PORT_FOR_icp-audit
+            hostname QRADAR_LOG_SOURCE_IDENTIFIER_FOR_icp-audit
+            protocol tcp
+            tls true
+            ca_file /fluentd/etc/tls/qradar.crt
+            packet_size 4096
+            program fluentd
+            <format>
+                @type single_value
+                message_key message
+            </format>
+        </store>`
+var qRadarConfigData2 = `
+    </match>`
+
+// DataSplunk defines the struct for splunk-hec-config
+type DataSplunk struct {
+	Value string `yaml:"splunkHEC.conf"`
+}
+
+// DataQRadar defines the struct for remote-syslog-config
+type DataQRadar struct {
+	Value string `yaml:"remoteSyslog.conf"`
+}
+
+// BuildConfigMap returns a ConfigMap object
+func BuildConfigMap(instance *operatorv1alpha1.AuditLogging, name string) (*corev1.ConfigMap, error) {
+	reqLogger := log.WithValues("ConfigMap.Namespace", InstanceNamespace, "ConfigMap.Name", name)
+	metaLabels := LabelsForMetadata(FluentdName)
+	dataMap := make(map[string]string)
+	var err error
+	switch name {
+	case FluentdDaemonSetName + "-" + ConfigName:
+		dataMap[enableAuditLogForwardKey] = strconv.FormatBool(instance.Spec.Fluentd.EnableAuditLoggingForwarding)
+		type Data struct {
+			Value string `yaml:"fluent.conf"`
+		}
+		d := Data{}
+		err = yaml.Unmarshal([]byte(fluentdMainConfigData), &d)
+		dataMap[fluentdConfigKey] = d.Value
+	case FluentdDaemonSetName + "-" + SourceConfigName:
+		type DataS struct {
+			Value string `yaml:"source.conf"`
+		}
+		ds := DataS{}
+		var result string
+		if instance.Spec.Fluentd.JournalPath != "" {
+			result = sourceConfigData1 + instance.Spec.Fluentd.JournalPath + sourceConfigData2
+		} else {
+			result = sourceConfigData1 + defaultJournalPath + sourceConfigData2
+		}
+		p := strconv.Itoa(defaultHTTPPort)
+		result += sourceConfigData3 + p + sourceConfigData4
+		err = yaml.Unmarshal([]byte(result), &ds)
```

Comment:
Same comment here about checking err.

Reply:
Unmarshal returns an error https://godoc.org/gopkg.in/yaml.v2#Unmarshal